into-base36-id/src/main.rs

```diff
@@ -1,19 +1,39 @@
 use ed25519_dalek::*;
 use failure::*;
 use hpos_config_core::*;
+use hpos_config_seed_bundle_explorer::unlock;
+use std::path::PathBuf;
+use structopt::StructOpt;
 
-use std::io::stdin;
+#[tokio::main]
+async fn main() -> Fallible<()> {
+    #[derive(StructOpt)]
+    struct Cli {
+        #[structopt(long = "config-path")]
+        /// The path to the hpos-config file
+        config_path: PathBuf,
+        #[structopt(long = "password")]
+        /// The password to unlock the device-bundle
+        password: String,
+    }
 
-fn main() -> Fallible<()> {
-    match serde_json::from_reader(stdin())? {
+    let Cli {
+        config_path,
+        password,
+        ..
+    } = Cli::from_args();
+    use std::fs::File;
+    let config_file = File::open(config_path).context("failed to open file")?;
+    match serde_json::from_reader(config_file)? {
         Config::V1 { seed, .. } => {
             let secret_key = SecretKey::from_bytes(&seed)?;
             let public_key = PublicKey::from(&secret_key);
             println!("{}", public_key::to_base36_id(&public_key));
         }
-        Config::V2 { encrypted_key, .. } => {
-            let key = Config::decode_key(&encrypted_key)?;
-            println!("{}", public_key::to_base36_id(&key.public));
+        Config::V2 { device_bundle, .. } => {
+            // take in password
+            let Keypair { public, .. } = unlock(&device_bundle, Some(password)).await.unwrap();
```

Comment:
I have to make a bit more research on the right way to handle an error in this case.

Reply:
Here is what I think. Because `into-base36-id` is a CLI binary in case of error it should:
- [ ]  exit with non-zero status
- [ ] print to stdout the reason for easier debugging

Both propagating Err with ? up to `main()` and panic from `unwrap()` will make program exit with status code 1. So first one is ✅ 

`context()` adds human-readible  message so every time it's used on Result second condition is also met. What we are missing in line 35 is exactly that - a context. So IMHO instead of `.unwrap()` we should `.context("Unable to unlock device bundle")`

Reply:
Unfortunately I just realized that `unlock()` return type is `Result<T, String>` and `the trait 'std::error::Error' is not implemented for 'std::string::String'`.

Reply:
I think that this should be changed.

Reply:
Implementation of custom error enum in fc2cc4f looks great!

Reply:
Added error handling in CLI bins in #74